### ISSUE TITLE
Single fetch: thrown response stub passed to `handleError`

### ISSUE THREAD

**Issue by haines** [remix-run/remix#9358]
### Reproduction

https://stackblitz.com/edit/remix-run-remix-afzd46?file=app%2Froutes%2F_index.tsx

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @remix-run/dev: * => 2.9.1 
    @remix-run/node: * => 2.9.1 
    @remix-run/react: * => 2.9.1 
    @remix-run/serve: * => 2.9.1 
    vite: ^5.1.0 => 5.2.10
```

### Used Package Manager

npm

### Expected Behavior

It should be possible to short-circuit loaders and actions by throwing the response stub ([docs](https://remix.run/docs/en/main/guides/single-fetch#headers)).

It should behave the same as throwing a `Response` object (it does not call `handleError`) ([docs](https://remix.run/docs/en/main/file-conventions/entry.server#thrown-responses)).

### Actual Behavior

`handleError` is called with the response stub.

**Comment by brophdawg11** [comment]
This fix is available in `2.9.2-pre.1` if you'd like to give it a shot and see if it resolves your issue!  Here's the draft release notes: https://github.com/remix-run/remix/blob/release-next/CHANGELOG.md#v292

**Comment by brophdawg11** [comment]
`2.9.2` is released - https://github.com/remix-run/remix/releases/tag/remix%402.9.2
